### PR TITLE
Refactor(eos_cli_config_gen): Deprecate Upper case letter Vxlan1 to vxlan1 for vxlan_interface schema

### LINKED ISSUE #4244: Deprecate schema keys with Upper case letter
### Enhancement summary

Find any keys with uppercase  in eos_cli_config_gen and eos_designs schemas and deprecate in favor of lower case. 

Ex. `vxlan_interface.Vxlan1` -> `vxlan_interface.vxlan1`

### Which component of AVD is impacted

eos_designs, eos_cli_config_gen

### Use case example

Ex. `vxlan_interface.Vxlan1` -> `vxlan_interface.vxlan1`

### Describe the solution you would like

deprecate the keys in favor of lower case. 

### Describe alternatives you have considered

_No response_

### Additional context

_No response_

### Contributing Guide

- [X] I agree to follow this project's Code of Conduct

### PR DESCRIPTION
**Pull request by MaheshGSLAB**
## Change Summary

Deprecate Upper case letter Vxlan1 to vxlan1 for vxlan_interface schema

## Related Issue(s)

Fixes #4244 

## Component(s) name

`arista.avd.eos_cli_config_gen`

## Proposed changes
Deprecate Upper case letter Vxlan1 to vxlan1 for vxlan_interface schema

## How to test
Run molecule

### User Checklist

- N/A

### Repository Checklist

- [x] My code has been rebased from devel before I start
- [x] I have read the [**CONTRIBUTING**](https://avd.sh/en/latest/docs/contribution/overview.html) document.
- [x] My change requires a change to the documentation and documentation have been updated accordingly.
- [x] I have updated [molecule CI](https://github.com/aristanetworks/avd/tree/devel/ansible_collections/arista/avd/molecule) testing accordingly. (check the box if not applicable)

### CODE REVIEW

File: python-avd/pyavd/_eos_cli_config_gen/j2templates/documentation/vxlan-interface.j2 [hunk 1]
```diff
@@ -5,76 +5,77 @@
 #}
 {# j2lint: disable=V1 #}
 {# doc - vxlan interface #}
-{% if vxlan_interface.Vxlan1 is arista.avd.defined %}
+{% set vxlan_config = vxlan_interface.vxlan1 if vxlan_interface.vxlan1 is arista.avd.defined else vxlan_interface.Vxlan1 %}
```
Comment (ClausHolbechArista) [inline python-avd/pyavd/_eos_cli_config_gen/j2templates/documentation/vxlan-interface.j2]:
```suggestion
{% set vxlan_config = vxlan_interface.vxlan1 | arista.avd.default(vxlan_interface.Vxlan1) %}
```

File: python-avd/pyavd/_eos_cli_config_gen/j2templates/eos/vxlan-interface.j2 [hunk 2]
```diff
@@ -5,84 +5,85 @@
 #}
 {# j2lint: disable=V1 #}
 {# eos - vxlan interface #}
-{% if vxlan_interface.Vxlan1 is arista.avd.defined %}
+{% set vxlan_config = vxlan_interface.vxlan1 if vxlan_interface.vxlan1 is arista.avd.defined else vxlan_interface.Vxlan1 %}
```
Comment (ClausHolbechArista) [inline python-avd/pyavd/_eos_cli_config_gen/j2templates/eos/vxlan-interface.j2]:
```suggestion
{% set vxlan_config = vxlan_interface.vxlan1 | arista.avd.default(vxlan_interface.Vxlan1) %}
```

**Review by ClausHolbechArista — DISMISSED** [review]

**Comment by ClausHolbechArista** [comment]
This deprecation will need to wait for 5.0.0 since it would trigger deprecation warnings for eos_designs run. In 5.0.0 we should change output of eos_designs to use this new key and deprecate the old for removal in 6.0.0. Too bad.

**Review by ClausHolbechArista — APPROVED** [review]

**Review by Shivani-gslab — APPROVED** [review]
LGTM!

**Pull request merged**